### PR TITLE
Add limits+coredis to mypy

### PR DESCRIPTION
**Pull request by mskarlin**
I forgot to add to mypy, not it's added. It revealed a couple of type signatures to clean up.

### CODE REVIEW

File: paperqa/rate_limiter.py [hunk 1]
```diff
@@ -301,7 +310,7 @@ async def try_acquire(
         """Returns when the limit is satisfied for the namespace_and_key.
 
         Args:
-            namespace_and_key (:obj:`tuple[str, str | MatchAllInputs]`): is
+            namespace_and_key (:obj:`tuple[str, str]`): is
```
Comment (jamesbraza) [inline paperqa/rate_limiter.py:313]: What is this `:obj:` syntax here? Normally it's good to not restate types if they're in the docstring already

**Review by jamesbraza — APPROVED** [review]
Thanks for doing this!

File: paperqa/rate_limiter.py (hunk 1 above)
Comment (mskarlin) [reply paperqa/rate_limiter.py:313]: it's in the style guide for the google doc strings, I think for sphinx compatibility

Comment (jamesbraza) [reply paperqa/rate_limiter.py:313]: Fwiw, either Sphinx autodoc or napoleon (I can't remember which) will pull in type hints if they are there

https://google.github.io/styleguide/pyguide.html#383-functions-and-methods note that it does not restate type hints in `Args` or `Returns`

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html in this one, note there's not type hints at all

Anyways, moral of the story I have seen it where people later update or expand type hints, but then don't update the repeated version in the docstring `Args` or `Returns`

**Pull request merged**